src/components/button/button.module.css

```diff
@@ -15,6 +15,7 @@
   font-size: var(--font-size);
   justify-content: center;
   line-height: var(--line-height);
+  min-height: var(--min-height);
```

Comment:
I've added this to match the [HDS implementation](). Without it, it seems that `small` size buttons are smaller than the expected `28px` height.

Happy to drop these changes from this PR and handle later - aim here was to match the design spec exactly, but I think it could make sense to get very close (does not require these changes) and iron out the `button` details more intentionally (if they don't seem quite right).

Reply:
Which buttons are you seeing this behavior for? I checked a couple at https://developer.hashi-mktg.com/swingset/components/button in Chrome, Safari, and Firefox and haven't found the same thing you have yet.

Reply:
Did some investigation, and turns out the `line-height` for the `small` variant was changed 10 days ago! 😄
https://github.com/hashicorp/design-system/commit/822154e4b84447d93a574c5845754d5beb856788

The new value is `0.875rem` (as you have it below), so you can just add that and remove the `min-height` stuff since it isn't needed.

Reply:
I think the `min-height` might still be needed - I'm seeing a height of `26px` when I remove it:

<img width="827" alt="CleanShot 2022-04-07 at 17 27 02@2x" src="https://user-images.githubusercontent.com/4624598/162320909-7f6d88b1-137e-43e5-baf4-cf08bec02d2c.png">

vs the expected `28px` when I have it in place:

<img width="828" alt="CleanShot 2022-04-07 at 17 27 19@2x" src="https://user-images.githubusercontent.com/4624598/162321080-f8b591a7-5558-48ec-bc5a-5c599cce5af8.png">

Let me know if that’s acceptable, happy to adjust if there’s another approach we want to take to meet the design spec 🙇‍♂️ Or also happy to discard the change if we don't mind the `2px` difference.

Reply:
Took another look at the padding in small buttons here: https://design-system-components-hashicorp.vercel.app/components/button, and found that our top/bottom padding is off by 1! That update was made in the same one as linked above for the `line-height` change and I didn't notice it until now.

So request is:

- Remove `--min-height` (For context: we shouldn't be setting any `height` properties because it makes buttons less responsive with font size. Buttons (or containers with text) should always grow with the font size.)
- Update the padding top/bottom for the `.small` variant to from `--padding-top-bottom: calc(6px - var(--border-width));` to `--padding-top-bottom: calc(7px - var(--border-width));`

Reply:
☝️ sounds good, just made this change ✅

For context on why I added `min-height`: I was going off the HDS implementation, which seems to use `min-height` (I think unlike `height` this still allows buttons to grow with the font size).

Reply:
That's true, but unfortunately `min-height` doesn't work in the other direction as far as I recall, like when users set their font size to something smaller than whatever `1rem` comes out to be. What'll happen is the text will shrink but the button will stop at whatever `min-height` is defined to be, and then the text will no longer be vertically aligned. That adds a need for `min-height` to be set in `rem` units, which is complex in our case because we're using `px` units for `padding` (that's another topic lol), so then we'd need to use `calc` to do `px` + `rem` math and IMO that's all more complex than ensuring height is correct in both font-size directions using just the `line-height` and `padding` properties.